pkg/interface/publish/src/js/components/lib/sidebar.js

```diff
@@ -6,62 +6,151 @@ export class Sidebar extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      sort: "oldest"
+      sort: "oldest",
+      sortedBooks: new Map()
     }
+    this.sort = this.sort.bind(this);
+    this.sortChange = this.sortChange.bind(this);
   }
 
-  render() {
-    const { props, state } = this;
+  componentDidMount() {
+    this.sort();
+  }
 
-    let activeClasses = (this.props.active === "sidebar") ? " " : "dn-s ";
+  componentDidUpdate(prevProps, prevState) {
+    if ((prevState.sort !== this.state.sort) || (prevProps !== this.props)) {
+      this.sort();
+    }
+  }
 
-    let hiddenClasses = true;
+  sort() {
+    let { props, state } = this;
+    let notebooks = new Map();
```

Comment:
sanity check these to make sure a[1] and b[1] exist before checking properties within them. program defensively!